### PR TITLE
3714 Validate sample sex in cg add relationship Command

### PR DESCRIPTION
**Pull request by RasmusBurge-CG**
### Added

- A check to check sex for father and mother for the CLI command `add relationship`
- Added a test to test if sex is incorrect, mother having sex = male, etc. 

### Changed

- Command `add relationship` will now be sex sensitive, and mother needs to have sex female, father needs to have sex male. 

### Fixed

- This will fix situations when prod needs to add a relationship manually.

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [x] Do test to assign parents with incorrect sex

### Expected test outcome

- [x] Check that you can assign sample that are correct.
- [x] Get an error if you add incorrect. 

```
[11:22] [hiseq.clinical@hasta:~] [S_base]  $ cg add relationship -f SampleX -s affected gentlewren SampleY
SampleX: father is not male
Aborted!
[11:28] [hiseq.clinical@hasta:~] [S_base]  $ cg add relationship -m SampleX -s affected gentlewren SampleY
SampleX: mother is not female
Aborted!
```

## Review

- [x] Tests executed by @RasmusBurge-CG 
- [x] "Merge and deploy" approved by @islean @diitaz93 
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/cli/add.py [hunk 1]
```diff
@@ -337,12 +337,20 @@ def link_sample_to_case(
...
+            raise click.Abort
+
     if father_id:
         father: Sample = status_db.get_sample_by_internal_id(internal_id=father_id)
         if father is None:
             LOG.error(f"{father_id}: father not found")
             raise click.Abort
 
+        if father.sex != Sex.MALE:
+            LOG.error(f"{father}: father is not {Sex.MALE}")
```
Comment (islean) [inline cg/cli/add.py:351]:
```suggestion
            LOG.error(f"{father_id}: father is not {Sex.MALE}")
```

**Review by islean — APPROVED** [review]
Looks good!

**Review by diitaz93 — APPROVED** [review]
Looks good 👍

**Comment by RasmusBurge-CG** [comment]
```
_cg -t cg -b master
Collecting git+https://github.com/Clinical-Genomics/cg@master
  Cloning https://github.com/Clinical-Genomics/cg (to revision master) to ./tmp/pip-req-build-gbb9cikw
  Running command git clone --quiet https://github.com/Clinical-Genomics/cg /home/hiseq.clinical/tmp/pip-req-build-gbb9cikw
  Resolved https://github.com/Clinical-Genomics/cg to commit a705ddebcd7d86b56ac62c1a7d9f8d2846419fcd
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: cg
  Building wheel for cg (pyproject.toml) ... done
  Created wheel for cg: filename=cg-64.4.19-py3-none-any.whl size=7845494 sha256=e932e79e67e01dc7e846de56a0de656d51a42d9d00cbacf3d1bb754cc5cf45c8
  Stored in directory: /home/hiseq.clinical/tmp/pip-ephem-wheel-cache-_k5z59o8/wheels/70/a8/7b/4abe0a2bf6472156279388f2b6206b4edcafc346e623de2208
Successfully built cg
Installing collected packages: cg
  Attempting uninstall: cg
    Found existing installation: cg 64.4.19
    Uninstalling cg-64.4.19:
      Successfully uninstalled cg-64.4.19
Successfully installed cg-64.4.19
Collecting git+https://github.com/Clinical-Genomics/cg@master
  Cloning https://github.com/Clinical-Genomics/cg (to revision master) to ./tmp/pip-req-build-uqg9lrem
  Running command git clone --quiet https://github.com/Clinical-Genomics/cg /home/hiseq.clinical/tmp/pip-req-build-uqg9lrem
  Resolved https://github.com/Clinical-Genomics/cg to commit a705ddebcd7d86b56ac62c1a7d9f8d2846419fcd
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: CacheControl in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (0.14.0)
Requirement already satisfied: Flask in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (3.0.3)
Requirement already satisfied: Flask-Admin in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (1.6.1)
Requirement already satisfied: Flask-CORS in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (4.0.1)
Requirement already satisfied: Flask-Dance in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (7.1.0)
Requirement already satisfied: Flask-WTF in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (1.2.1)
Requirement already satisfied: Jinja2>=3.1.3 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (3.1.4)
Requirement already satisfied: MarkupSafe in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.1.5)
Requirement already satisfied: PyMySQL in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (1.1.1)
Requirement already satisfied: PyYAML in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (6.0.1)
Requirement already satisfied: SQLAlchemy in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.0.30)
Requirement already satisfied: WTForms in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (3.1.2)
Requirement already satisfied: alembic in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (1.13.1)
Requirement already satisfied: blinker in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (1.8.2)
Requirement already satisfied: cachetools in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (5.3.3)
Requirement already satisfied: click in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (8.1.7)
Requirement already satisfied: coloredlogs in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (15.0.1)
Requirement already satisfied: cryptography in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (42.0.8)
Requirement already satisfied: email-validator<3.0.0,>=2.2.0 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.2.0)
Requirement already satisfied: genologics in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (1.0.0)
Requirement already satisfied: google-auth in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.30.0)
Requirement already satisfied: gunicorn in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (22.0.0)
Requirement already satisfied: housekeeper>=4.11.3 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (4.13.1)
Requirement already satisfied: lxml in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (5.2.2)
Requirement already satisfied: marshmallow in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (3.21.3)
Requirement already satisfied: openpyxl in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (3.1.3)
Requirement already satisfied: packaging in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (24.1)
Requirement already satisfied: pandas in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.2.2)
Requirement already satisfied: paramiko in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (3.4.0)
Requirement already satisfied: petname in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.6)
Requirement already satisfied: psutil in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (5.9.8)
Requirement already satisfied: pydantic in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.7.3)
Requirement already satisfied: pydantic-settings<3.0.0,>=2.3.3 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.3.3)
Requirement already satisfied: python-dateutil in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.9.0.post0)
Requirement already satisfied: requests in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.32.3)
Requirement already satisfied: tabulate in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (0.9.0)
Requirement already satisfied: typing_extensions in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (4.12.2)
Requirement already satisfied: urllib3 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (2.2.1)
Requirement already satisfied: werkzeug in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cg==64.4.19) (3.0.3)
Requirement already satisfied: dnspython>=2.0.0 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from email-validator<3.0.0,>=2.2.0->cg==64.4.19) (2.6.1)
Requirement already satisfied: idna>=2.0.0 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from email-validator<3.0.0,>=2.2.0->cg==64.4.19) (3.7)
Requirement already satisfied: rich in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from housekeeper>=4.11.3->cg==64.4.19) (13.7.1)
Requirement already satisfied: python-dotenv>=0.21.0 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from pydantic-settings<3.0.0,>=2.3.3->cg==64.4.19) (1.0.1)
Requirement already satisfied: annotated-types>=0.4.0 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from pydantic->cg==64.4.19) (0.7.0)
Requirement already satisfied: pydantic-core==2.18.4 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from pydantic->cg==64.4.19) (2.18.4)
Requirement already satisfied: Mako in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from alembic->cg==64.4.19) (1.3.5)
Requirement already satisfied: greenlet!=0.4.17 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from SQLAlchemy->cg==64.4.19) (3.0.3)
Requirement already satisfied: msgpack<2.0.0,>=0.5.2 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from CacheControl->cg==64.4.19) (1.0.8)
Requirement already satisfied: charset-normalizer<4,>=2 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from requests->cg==64.4.19) (3.3.2)
Requirement already satisfied: certifi>=2017.4.17 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from requests->cg==64.4.19) (2024.6.2)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from coloredlogs->cg==64.4.19) (10.0)
Requirement already satisfied: cffi>=1.12 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cryptography->cg==64.4.19) (1.16.0)
Requirement already satisfied: itsdangerous>=2.1.2 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from Flask->cg==64.4.19) (2.2.0)
Requirement already satisfied: oauthlib>=3.2 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from Flask-Dance->cg==64.4.19) (3.2.2)
Requirement already satisfied: requests-oauthlib>=1.0.0 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from Flask-Dance->cg==64.4.19) (2.0.0)
Requirement already satisfied: urlobject in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from Flask-Dance->cg==64.4.19) (2.4.3)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from google-auth->cg==64.4.19) (0.4.0)
Requirement already satisfied: rsa<5,>=3.1.4 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from google-auth->cg==64.4.19) (4.9)
Requirement already satisfied: et-xmlfile in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from openpyxl->cg==64.4.19) (1.1.0)
Requirement already satisfied: numpy>=1.23.2 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from pandas->cg==64.4.19) (1.26.4)
Requirement already satisfied: pytz>=2020.1 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from pandas->cg==64.4.19) (2024.1)
Requirement already satisfied: tzdata>=2022.7 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from pandas->cg==64.4.19) (2024.1)
Requirement already satisfied: six>=1.5 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from python-dateutil->cg==64.4.19) (1.16.0)
Requirement already satisfied: bcrypt>=3.2 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from paramiko->cg==64.4.19) (4.1.3)
Requirement already satisfied: pynacl>=1.5 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from paramiko->cg==64.4.19) (1.5.0)
Requirement already satisfied: pycparser in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from cffi>=1.12->cryptography->cg==64.4.19) (2.22)
Requirement already satisfied: pyasn1<0.7.0,>=0.4.6 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from pyasn1-modules>=0.2.1->google-auth->cg==64.4.19) (0.6.0)
Requirement already satisfied: markdown-it-py>=2.2.0 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from rich->housekeeper>=4.11.3->cg==64.4.19) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from rich->housekeeper>=4.11.3->cg==64.4.19) (2.18.0)
Requirement already satisfied: mdurl~=0.1 in /home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages (from markdown-it-py>=2.2.0->rich->housekeeper>=4.11.3->cg==64.4.19) (0.1.2)
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical
done.
Log deploy... done.
cg, version 64.4.19
[11:45] [hiseq.clinical@hasta:~] [S_base]  $ client_loop: send disconnect: Broken pipe
(base) rasmusburge@rasmusbmba ~ % hasta
Last login: Mon Nov 11 09:51:33 2024 from sll-vpn-nod3.scilifelab.se
Rocks 7.0 (Manzanita)
Profile built 14:16 29-Aug-2018

Kickstarted 15:11 29-Aug-2018
[rasmus.burge@hasta:~] [base] $ sudo -iu hiseq.clinical
up
[12:04] [hiseq.clinical@hasta:~] [base]  $ up
[12:04] [hiseq.clinical@hasta:~] [P_base]  $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-prod.sh -e P_cg -t cg -b master
Collecting git+https://github.com/Clinical-Genomics/cg@master
  Cloning https://github.com/Clinical-Genomics/cg (to revision master) to ./tmp/pip-req-build-agucdvu8
  Running command git clone --quiet https://github.com/Clinical-Genomics/cg /home/hiseq.clinical/tmp/pip-req-build-agucdvu8
  Resolved https://github.com/Clinical-Genomics/cg to commit 2d03ee2a5f79a4e87596d3185ff28606f916305a
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: cg
  Building wheel for cg (pyproject.toml) ... done
  Created wheel for cg: filename=cg-64.4.20-py3-none-any.whl size=7845757 sha256=fe789b194c254aa38bc7aed9f1b4febbccde4d3c1cc0d37c0905be4c30819622
  Stored in directory: /home/hiseq.clinical/tmp/pip-ephem-wheel-cache-5yub7z7p/wheels/70/a8/7b/4abe0a2bf6472156279388f2b6206b4edcafc346e623de2208
Successfully built cg
Installing collected packages: cg
  Attempting uninstall: cg
    Found existing installation: cg 64.4.19
    Uninstalling cg-64.4.19:
      Successfully uninstalled cg-64.4.19
Successfully installed cg-64.4.20
Collecting git+https://github.com/Clinical-Genomics/cg@master
  Cloning https://github.com/Clinical-Genomics/cg (to revision master) to ./tmp/pip-req-build-xpgbsxw9
  Running command git clone --quiet https://github.com/Clinical-Genomics/cg /home/hiseq.clinical/tmp/pip-req-build-xpgbsxw9
  Resolved https://github.com/Clinical-Genomics/cg to commit 2d03ee2a5f79a4e87596d3185ff28606f916305a
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: CacheControl in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (0.14.0)
Requirement already satisfied: Flask in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (3.0.2)
Requirement already satisfied: Flask-Admin in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (1.6.1)
Requirement already satisfied: Flask-CORS in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (4.0.0)
Requirement already satisfied: Flask-Dance in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (7.1.0)
Requirement already satisfied: Flask-WTF in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (1.2.1)
Requirement already satisfied: Jinja2>=3.1.3 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (3.1.3)
Requirement already satisfied: MarkupSafe in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.1.5)
Requirement already satisfied: PyMySQL in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (1.1.0)
Requirement already satisfied: PyYAML in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (6.0.1)
Requirement already satisfied: SQLAlchemy in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.0.29)
Requirement already satisfied: WTForms in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (3.1.2)
Requirement already satisfied: alembic in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (1.13.1)
Requirement already satisfied: blinker in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (1.7.0)
Requirement already satisfied: cachetools in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (5.3.3)
Requirement already satisfied: click in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (8.1.7)
Requirement already satisfied: coloredlogs in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (15.0.1)
Requirement already satisfied: cryptography in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (42.0.5)
Requirement already satisfied: email-validator<3.0.0,>=2.2.0 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.2.0)
Requirement already satisfied: genologics in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (1.0.0)
Requirement already satisfied: google-auth in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.29.0)
Requirement already satisfied: gunicorn in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (21.2.0)
Requirement already satisfied: housekeeper>=4.11.3 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (4.11.4)
Requirement already satisfied: lxml in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (5.1.0)
Requirement already satisfied: marshmallow in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (3.21.1)
Requirement already satisfied: openpyxl in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (3.1.2)
Requirement already satisfied: packaging in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (24.0)
Requirement already satisfied: pandas in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.2.1)
Requirement already satisfied: paramiko in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (3.4.0)
Requirement already satisfied: petname in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.6)
Requirement already satisfied: psutil in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (5.9.8)
Requirement already satisfied: pydantic in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.7.4)
Requirement already satisfied: pydantic-settings<3.0.0,>=2.3.3 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.3.3)
Requirement already satisfied: python-dateutil in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.9.0.post0)
Requirement already satisfied: requests in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.31.0)
Requirement already satisfied: tabulate in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (0.9.0)
Requirement already satisfied: typing_extensions in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (4.10.0)
Requirement already satisfied: urllib3 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (2.2.1)
Requirement already satisfied: werkzeug in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cg==64.4.20) (3.0.1)
Requirement already satisfied: dnspython>=2.0.0 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from email-validator<3.0.0,>=2.2.0->cg==64.4.20) (2.6.1)
Requirement already satisfied: idna>=2.0.0 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from email-validator<3.0.0,>=2.2.0->cg==64.4.20) (3.6)
Requirement already satisfied: rich in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from housekeeper>=4.11.3->cg==64.4.20) (13.7.1)
Requirement already satisfied: python-dotenv>=0.21.0 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from pydantic-settings<3.0.0,>=2.3.3->cg==64.4.20) (1.0.1)
Requirement already satisfied: annotated-types>=0.4.0 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from pydantic->cg==64.4.20) (0.6.0)
Requirement already satisfied: pydantic-core==2.18.4 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from pydantic->cg==64.4.20) (2.18.4)
Requirement already satisfied: Mako in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from alembic->cg==64.4.20) (1.3.2)
Requirement already satisfied: greenlet!=0.4.17 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from SQLAlchemy->cg==64.4.20) (3.0.3)
Requirement already satisfied: msgpack<2.0.0,>=0.5.2 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from CacheControl->cg==64.4.20) (1.0.8)
Requirement already satisfied: charset-normalizer<4,>=2 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from requests->cg==64.4.20) (3.3.2)
Requirement already satisfied: certifi>=2017.4.17 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from requests->cg==64.4.20) (2024.2.2)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from coloredlogs->cg==64.4.20) (10.0)
Requirement already satisfied: cffi>=1.12 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cryptography->cg==64.4.20) (1.16.0)
Requirement already satisfied: itsdangerous>=2.1.2 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from Flask->cg==64.4.20) (2.1.2)
Requirement already satisfied: oauthlib>=3.2 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from Flask-Dance->cg==64.4.20) (3.2.2)
Requirement already satisfied: requests-oauthlib>=1.0.0 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from Flask-Dance->cg==64.4.20) (2.0.0)
Requirement already satisfied: urlobject in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from Flask-Dance->cg==64.4.20) (2.4.3)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from google-auth->cg==64.4.20) (0.3.0)
Requirement already satisfied: rsa<5,>=3.1.4 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from google-auth->cg==64.4.20) (4.9)
Requirement already satisfied: et-xmlfile in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from openpyxl->cg==64.4.20) (1.1.0)
Requirement already satisfied: numpy<2,>=1.23.2 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from pandas->cg==64.4.20) (1.26.4)
Requirement already satisfied: pytz>=2020.1 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from pandas->cg==64.4.20) (2024.1)
Requirement already satisfied: tzdata>=2022.7 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from pandas->cg==64.4.20) (2024.1)
Requirement already satisfied: six>=1.5 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from python-dateutil->cg==64.4.20) (1.16.0)
Requirement already satisfied: bcrypt>=3.2 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from paramiko->cg==64.4.20) (4.1.2)
Requirement already satisfied: pynacl>=1.5 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from paramiko->cg==64.4.20) (1.5.0)
Requirement already satisfied: pycparser in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from cffi>=1.12->cryptography->cg==64.4.20) (2.21)
Requirement already satisfied: pyasn1<0.6.0,>=0.4.6 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from pyasn1-modules>=0.2.1->google-auth->cg==64.4.20) (0.5.1)
Requirement already satisfied: markdown-it-py>=2.2.0 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from rich->housekeeper>=4.11.3->cg==64.4.20) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from rich->housekeeper>=4.11.3->cg==64.4.20) (2.17.2)
Requirement already satisfied: mdurl~=0.1 in /home/proj/production/bin/miniconda3/envs/P_cg/lib/python3.11/site-packages (from markdown-it-py>=2.2.0->rich->housekeeper>=4.11.3->cg==64.4.20) (0.1.2)
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical
done.
Log deploy... done.
cg, version 64.4.20
```

**Pull request merged**